Avoid data-dependent errors in NJT tests via capture_scalar_outputs=True

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #144889
* __->__ #144588
* #144587
* #144586

Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

There are several xfails related to data-dependent errors in torch.compile. This PR sets `torch._dynamo.config.capture_scalar_outputs=True` to avoid these, which tends to exercise unbacked SymInt logic and will require `torch._check()`-related fixes.

jbschlosser committed:
Avoid data-dependent errors in NJT tests via capture_scalar_outputs=True

[ghstack-poisoned]

soulitzer reviewed (approved):
Nice clean up!

soulitzer commented on test/test_nestedtensor.py:
```diff
@@ -8654,11 +8584,13 @@ def test_backward(self, device, dtype, op):
     )
+    @torch._dynamo.config.patch(capture_dynamic_output_shape_ops=True)
+    # needed to avoid "data dependent operator: aten._local_scalar_dense.default"
+    @torch._dynamo.config.patch(capture_scalar_outputs=True)
```
Yep saw these changes as part of the narrow PR as well haha

Though one reason for not doing it by default for all ops is if we want to get signal on which ops need which flags to work with compile, but maybe not worth the hassle.

jbschlosser replied:
I think what I want to do is have a section of the upcoming NJT docs dedicated to torch.compile-based stuff, and mention the trick to add this flag as needed for data-dependent operators.

jbschlosser committed:
Update on "Avoid data-dependent errors in NJT tests via capture_scalar_outputs=True"


Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

There are several xfails related to data-dependent errors in torch.compile. This PR sets `torch._dynamo.config.capture_scalar_outputs=True` to avoid these, which tends to exercise unbacked SymInt logic and will require `torch._check()`-related fixes.

[ghstack-poisoned]

jbschlosser committed:
Update on "Avoid data-dependent errors in NJT tests via capture_scalar_outputs=True"


Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

There are several xfails related to data-dependent errors in torch.compile. This PR sets `torch._dynamo.config.capture_scalar_outputs=True` to avoid these, which tends to exercise unbacked SymInt logic and will require `torch._check()`-related fixes.

[ghstack-poisoned]

jbschlosser committed:
Update on "Avoid data-dependent errors in NJT tests via capture_scalar_outputs=True"


Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

There are several xfails related to data-dependent errors in torch.compile. This PR sets `torch._dynamo.config.capture_scalar_outputs=True` to avoid these, which tends to exercise unbacked SymInt logic and will require `torch._check()`-related fixes.

[ghstack-poisoned]

jbschlosser committed:
Update on "Avoid data-dependent errors in NJT tests via capture_scalar_outputs=True"


Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

There are several xfails related to data-dependent errors in torch.compile. This PR sets `torch._dynamo.config.capture_scalar_outputs=True` to avoid these, which tends to exercise unbacked SymInt logic and will require `torch._check()`-related fixes.

[ghstack-poisoned]

jbschlosser committed:
Update on "Avoid data-dependent errors in NJT tests via capture_scalar_outputs=True"


Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

There are several xfails related to data-dependent errors in torch.compile. This PR sets `torch._dynamo.config.capture_scalar_outputs=True` to avoid these, which tends to exercise unbacked SymInt logic and will require `torch._check()`-related fixes.

[ghstack-poisoned]
